admittance_controller/include/admittance_controller/admittance_rule_impl.hpp

```diff
@@ -0,0 +1,444 @@
+  success &= kinematics_->calculate_link_transform(
+    reference_joint_state.positions, parameters_.kinematics.tip, trans_ref_.base_tip_);
+  success &= kinematics_->calculate_link_transform(
+    reference_joint_state.positions, parameters_.fixed_world_frame.frame.id,
+    trans_ref_.world_base_);
+  success &= kinematics_->calculate_link_transform(
+    reference_joint_state.positions, parameters_.ft_sensor.frame.id, trans_ref_.base_sensor_);
+  success &= kinematics_->calculate_link_transform(
+    reference_joint_state.positions, parameters_.gravity_compensation.frame.id,
+    trans_ref_.base_cog_);
+  success &= kinematics_->calculate_link_transform(
+    reference_joint_state.positions, parameters_.control.frame.id, trans_ref_.base_control_);
+
+  // get all current transforms
+  success &= kinematics_->calculate_link_transform(
+    current_joint_state.positions, parameters_.ft_sensor.frame.id, trans_.base_ft_);
+  success &= kinematics_->calculate_link_transform(
+    current_joint_state.positions, parameters_.kinematics.tip, trans_.base_tip_);
+  success &= kinematics_->calculate_link_transform(
+    current_joint_state.positions, parameters_.fixed_world_frame.frame.id, trans_.world_base_);
+  success &= kinematics_->calculate_link_transform(
+    current_joint_state.positions, parameters_.ft_sensor.frame.id, trans_.base_sensor_);
+  success &= kinematics_->calculate_link_transform(
+    current_joint_state.positions, parameters_.gravity_compensation.frame.id, trans_.base_cog_);
+  success &= kinematics_->calculate_link_transform(
+    current_joint_state.positions, parameters_.control.frame.id, trans_.base_control_);
+
+  return success;
+}
+
+// Update from reference joint states
+controller_interface::return_type AdmittanceRule::update(
+  const trajectory_msgs::msg::JointTrajectoryPoint & current_joint_state,
+  const geometry_msgs::msg::Wrench & measured_wrench,
+  const trajectory_msgs::msg::JointTrajectoryPoint & reference_joint_state,
+  const rclcpp::Duration & period, trajectory_msgs::msg::JointTrajectoryPoint & desired_joint_state)
+{
+  const double dt = period.seconds();
+
+  if (parameters_.enable_parameter_update_without_reactivation)
+  {
+    apply_parameters_update();
+  }
+
+  bool success = get_all_transforms(current_joint_state, reference_joint_state, admittance_state_);
+
+  // calculate needed rotations
+  Eigen::Matrix<double, 3, 3> rot_base_sensor = trans_.base_sensor_.rotation();
+  Eigen::Matrix<double, 3, 3> rot_world_base = trans_.world_base_.rotation();
+  Eigen::Matrix<double, 3, 3> rot_base_cog = trans_.base_cog_.rotation();
+  Eigen::Matrix<double, 3, 3> rot_base_control = trans_.base_control_.rotation();
+
+  // apply filter and update wrench_world_ vector
+  Eigen::Matrix<double, 3, 3> rot_world_sensor = rot_world_base * rot_base_sensor;
+  Eigen::Matrix<double, 3, 3> rot_world_cog = rot_world_base * rot_base_cog;
```

Comment:
```suggestion
  // apply filter and update wrench_world_ vector
  Eigen::Matrix<double, 3, 3> rot_world_sensor = trans_.world_base_.rotation() * trans_.base_sensor_.rotation();
  Eigen::Matrix<double, 3, 3> rot_world_cog = trans_.world_base_.rotation() * trans_.base_cog_.rotation();
```

Reply:
I would still go without temporary variables. They are not necessary if used only once.